doc/GUIDE.md

```diff
@@ -608,6 +608,66 @@ public class MyPoint extends Point { }
 Through static inheritance, all the methods in the `Point` class are available through the
 `MyPoint` class as well.
 
+### Array Types
+
+Jextract treats variables (global variables or struct/union fields) with an array type
+specially. For instance, if we have a header file with the following declaration:
+
+```c
+// mylib.h
+
+int FOO_ARRAY[3][5];
+```
+
+Jextract generates a few extra methods that are useful for working with arrays:
+
+```java
+// mylib_h.java
+
+public static SequenceLayout FOO_ARRAY$layout() { ... } // 1
+public static long[] FOO_ARRAY$dimensions() { ... } // 2
+
+public static MemorySegment FOO_ARRAY() { ... } // 3
+public static void FOO_ARRAY(MemorySegment varValue) { ... } // 3
+
+public static int FOO_ARRAY(long index0, long index1) { ... } // 4
+public static void FOO_ARRAY(long index0, long index1, int varValue) { ... } // 4
+```
+
+Jextract generates:
+
+1. a layout accessor, just like we have for a regular variable, but note that the return
+  type is [`SequenceLayout`].
+2. a `$dimensions` meta-deta accessor, which returns the _dimensions_ of the array type.
+  This method returns a `long[]` where each element represents the length of a dimension
+  of the array type. For instance, in the example `FOO_ARRAY` has a two dimension, whose
+  lengths are `3` and `5`, so the `FOO_ARRAY$dimensions` method will return a `long[]`
+  with two elements whose values are `3` and `5` in that order.
+3. a getter and setter pair for the array variable. Note that the getter replaces the usual
```

Comment:
Should we start from this?

Reply:
I maintained the source order, but we can re-order (I did that for the function example as well).